src/lib/apis/opensearch.ts

```diff
@@ -0,0 +1,26 @@
+import urljoin from "url-join"
+import { assign } from "lodash"
+import config from "config"
+import fetch from "node-fetch"
+
+const { OPENSEARCH_API_BASE } = config
+
+export const opensearch = async (
+  path,
+  _accessToken,
+  fetchOptions: any = {}
+) => {
+  const headers = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+  }
+
+  const response = await (
+    await fetch(
+      urljoin(OPENSEARCH_API_BASE, path),
+      assign({}, fetchOptions, { headers })
+    )
+  ).json()
```

Comment:
This feels like a nice spike-y approach; get it working here, eventually move it to where it needs to be (when those addition ecosystem (grav) features are required)